modules/react/common/lib/responsive/useResponsiveContainerStyles.ts

```diff
@@ -0,0 +1,137 @@
+like `min-width`. For example, if you want to apply styles from `medium` and up, then you would
+write those styles under `m`.
+ *
+ * @example
+ * ```tsx
+import {Flex, Box} from '@workday/canvas-kit-react/layout';
+import {
+  useResponsiveStyles,
+} from '@workday/canvas-kit-react/common';
+
+const containerResponsiveStyles = useResponsiveStyles(
+  {
+    flex: {
+      flexDirection: 'column',
+      padding: 'm',
+      depth: 1,
+      borderRadius: 'l',
+      zero: {
+        backgroundColor: 'Red',
+      },
+      s: {
+        backgroundColor: 'Orange',
+      },
+      m: {
+        backgroundColor: 'Yellow',
+      },
+      l: {
+        backgroundColor: 'Green',
+      },
+      xl: {
+        backgroundColor: 'Blue',
+      },
+    },
+    box: {
+      padding: 's',
+    },
+  },
+  containerWidth
+);
+
+return (
+  <ResponsiveContextProvider width={containerWidth}>
+    <Box ref={ref}>
+      <Flex {...containerResponsiveStyles.flex}>
+        <Box {...containerResponsiveStyles.box}>Hello World</Box>
+      </Flex>
+    </Box>
+  </ResponsiveContextProvider>
+);
+```
+ */
+export function useResponsiveContainerStyles<T extends ResponsiveCSSObject<T>>(
+  styles: ResponsiveCSSObject<T>,
+  width: number,
+  theme = {}
+) {
+  const canvasTheme = useTheme(theme);
+  const breakpoints = canvasTheme.canvas.breakpoints.values;
+  // scoped strictly within the breakpoint range (think of it as the sum of min-width + max-width)
+  const isZero = isWithinBreakpoint(width, 0, breakpoints.s);
+  const isSmall = isWithinBreakpoint(width, breakpoints.s, breakpoints.m);
+  const isMedium = isWithinBreakpoint(width, breakpoints.m, breakpoints.l);
+  const isLarge = isWithinBreakpoint(width, breakpoints.l, breakpoints.xl);
+  const isExtraLarge = isWithinBreakpoint(width, breakpoints.xl);
+  const responsiveStyles = {} as CSSObject<T>;
+
+  function getStyles(key: BreakpointKeys) {
+    const breakpointSize = breakpointResKeys.indexOf(key);
+    for (let i = 0; i <= breakpointSize; i++) {
+      const breakpointName = breakpointResKeys[i];
+      Object.keys(styles).forEach((classname) => {
+        const { zero, s, m, l, xl, ...base } = styles[classname as keyof T];
+        const breakpointStyles =
+          styles[classname as keyof T][breakpointName] ?? {};
+        const existingStyles = responsiveStyles[classname as keyof T] ?? {};
+        responsiveStyles[classname as keyof T] = {
+          ...base,
+          ...existingStyles,
+          ...breakpointStyles
+        };
+      });
+    }
+  }
+
+  // eslint-disable-next-line default-case
+  switch (true) {
+    case isZero: {
+      getStyles("zero");
+      break;
+    }
+    case isSmall: {
+      getStyles("s");
+      break;
+    }
+    case isMedium: {
+      getStyles("m");
+      break;
+    }
+    case isLarge: {
+      getStyles("l");
+      break;
+    }
+    case isExtraLarge: {
+      getStyles("xl");
+      break;
+    }
+  }
```

Comment:
How do you feel about function like this: 
```
const getSize = (width: number, breakpoints: any) => {
  const ranges: {[key: string ]: [number, number?]} = {
    'zero': [0, breakpoints.s],
    's': [breakpoints.s, breakpoints.m],
    'm': [breakpoints.m, breakpoints.l],
    'l': [breakpoints.l, breakpoints.xl],
    'xl': [breakpoints.xl]
  };
  const size = breakpointResKeys.find((size: BreakpointKeys) => isWithinBreakpoint(width, ...ranges[size]))

  return size as BreakpointKeys;
}

...

const size = getSize(width, breakpoints);
getStyles(size);

```

It allows to remove switch and lines 101-105 (because you use it only ones.

Reply:
I really like Raisa's feedback and I was going to suggest moving in a similar direction. I'll add to what Raisa suggested.

Do everything she suggested, but then inline `getStyles` since it's only called once. I'd rather only have a function if it's a pure function, ideally. It's not, and the work it does makes sense, but if you inline it, I think that makes things a little more straightforward to follow.

Reply:
You should go with what Raisa and I suggested, but if there was some reason not to, the switch pattern here could still be simplified in a way. The alternative recommendation I'd make is to have `responsiveStyles` completely contained inside of `getStyles()`, but then have `getStyles() return the value at the end. Then instead of switch/cases, you just have a series of `if` blocks and you can even inline the variables so there is less math going on, but honestly that's a micro-optimization that's completely optional. It would look lik:
```
if(isWithinBreakpoint(width, 0, breakpoints.s)) {
  return getStyles("zero");
}
// other four cases in their own if blocks after
```

Reply:
I agree and I'm going to implement Raisa's suggestions

Reply:
RC: I agree that there's some refactoring to do here. I would also like this function pulled out of the hook so it's only instantiated once.